### PR TITLE
Show preview button for spotlight/infobar in experiments

### PR DESCRIPTION
**Pull request by AllegroFox**
Enables the "Preview Link" cell for infobars and Spotlights in the experiments table. `encodeURIcomponent` should also help with some of the `btoa` issues we were having, and `_substituteLocalizations` comes from Limelight and should allow recipes that include localization objects to still be previewed. (`about:messagepreview` doesn't know about localization and will error out if it's included in the object.)

### CODE REVIEW

File: app/page.tsx
Comment (dmose) [inline app/page.tsx:24]: Let's put this into experimentUtils.ts and add tests for at least important stuff.

File: app/columns.tsx
Comment (dmose) [inline app/columns.tsx:274]: The other side of the fork (in the ASRouter table) is actually more up-to-date re-styling.  Can you move that code into a PreviewLinkButton component in its own file and then call it from both there and here?

File: app/page.tsx
Comment (dmose) [reply app/page.tsx:22]: Also, thanks for noticing this bug and fixing it here in the ASRouter table too!

File: app/page.tsx
Comment (dmose) [inline app/page.tsx:164]: Please indent these two last to lines, as they are hard to read as-is.

File: lib/experimentUtils.ts
Comment (dmose) [inline lib/experimentUtils.ts:76]: Please add a detailed description of what the function is supposed to do.

Since we're adding new code here, can you write at least a few tests for the most important cases?

File: lib/experimentUtils.ts
Comment (dmose) [inline lib/experimentUtils.ts:92]: A comment above this for loop would be helpful for readers.

**Review by dmose — CHANGES_REQUESTED** [review]
This looks great -- thanks for finding and fixing the encodeURIComponent bug in both places!  Just a few requests before landing-ready...

**Comment by dmose** [comment]
Also, check https://github.com/mozilla/skylight/actions/runs/8116013167/job/22185107242?pr=76 from the CI.  For some reason (and I see this too), certain type-checking errors get caught by Netlify's CI, and by the eslint integration in VS Code, but not by the "npm run dev" and "npm run lint" steps.  I have yet to track this down.

**Comment by dmose** [comment]
Looks like the errors are locally visible with `npm run build` but not with `npm run dev`.

**Comment by dmose** [comment]
Yuck: intentional setup by next for performance: https://github.com/vercel/next.js/discussions/13617.   Easy enough to work around, though.

File: app/page.tsx
Comment (dmose) [reply app/page.tsx]: Also, adding types for this function would be worthwhile, and might solve some of the complaints from the compiler in `npm run build`.

File: app/columns.tsx
Comment (AllegroFox) [reply app/columns.tsx:274]: Can do!

File: app/page.tsx
Comment (AllegroFox) [reply app/page.tsx]: done!

File: app/page.tsx
Comment (AllegroFox) [inline app/page.tsx:62]: This `[Object.keys(recipe.localizations)[0]]` thing is kinda gross but it means we correctly access the first locale inside the localization object. We'll probably want to add a dropdown component that allows us to choose a locale from the available ones, to pass to this function.

Comment (dmose) [reply app/page.tsx:62]: How about adding this in as an XXX comment?

Comment (AllegroFox) [reply app/page.tsx:62]: Will do :)

Comment (dmose) [reply app/page.tsx:62]: Also, please try to avoid requiring readers to make mental context switches inside of a single line of code.  I.e. split out the `_substituteLocalizations` function call to at least one other line and a temporary variable, maybe more, depending on how readable it feels.

Comment (dmose) [reply app/page.tsx:62]: Another way to win on readability here might be to add a 'getFirstLocalization' method in experimentUtils.ts.  Your call.

File: app/page.tsx
Comment (dmose) [inline app/page.tsx:86]: Outdent to match the context, please.

File: app/page.tsx
Comment (dmose) [inline app/page.tsx:130]: If you could revert the commenting here, that'd be great.

File: lib/experimentUtils.ts
Comment (dmose) [inline lib/experimentUtils.ts:89]: How about copying [the localizations type from nimbus shared](https://github.com/mozilla/nimbus-shared/blob/main/types/experiments.ts#L121-L131)?

I'm not quite sure what to do about the values typing. If there's something we could copy-paste, that'd be great.  If not, c'est la vie. Up to you...

File: lib/messageUtils.ts
Comment (dmose) [inline lib/messageUtils.ts:54]: It seems like this function would never actually return undefined, would it?

If not, can we remove it from the type signature?

**Review by dmose — APPROVED** [review]
Looks great; thanks for the nice work!  Just some small bits to tidy up before landing...

r=dmose once you've made those tweaks

File: app/page.tsx
Comment (AllegroFox) [reply app/page.tsx:86]: Looks like there was a weird combo of tabs and spaces here. Fixed!

File: lib/messageUtils.ts
Comment (AllegroFox) [reply lib/messageUtils.ts:54]: yep!

**Pull request merged**
